src/chartserver/cache.go

```diff
@@ -106,12 +107,12 @@ func (chc *ChartCache) PutChart(chart *ChartVersionDetails) {
 		switch chc.driverType {
 		case cacheDriverMem:
 			// Directly put object in
-			err = chc.cache.Put(chart.Metadata.Digest, chart, standardExpireTime)
+			err = chc.cache.Put(context.Background(), chart.Metadata.Digest, chart, standardExpireTime)
```

Comment:
I suggest define a top level context in the beginning of this function, and use it for following ctx parameters which needed.